sim/midas/src/main/scala/midas/passes/CoerceAsyncToSyncReset.scala

```diff
@@ -0,0 +1,34 @@
+// See LICENSE for license details
+
+package midas.passes
+
+import firrtl._
+import firrtl.annotations._
+import firrtl.ir._
+import firrtl.Mappers._
+import firrtl.Utils.BoolType
+
+/**
+  * Replaces all AsyncResets and Reset types with Bools (synchronous reset).
+  *
+  */
+
+object CoerceAsyncToSyncReset extends firrtl.Transform {
+  override def name = "[Golden Gate] Coerce AsyncReset to SyncReset"
+  def inputForm = LowForm
+  def outputForm = LowForm
+
+  private def onType(t: Type): Type = t match {
+    case AsyncResetType => BoolType
+    case ResetType => BoolType
+    case o => o
+  }
+  private def onExpr(e: Expression): Expression = e.map(onExpr).map(onType) match {
+    case p@DoPrim(PrimOps.AsAsyncReset, Seq(arg), _, _) => arg
```

Comment:
```suggestion
    case p @ DoPrim(PrimOps.AsAsyncReset, Seq(arg), _, _) => arg
```

Reply:
Technically the map(onType) would have already made `p.tpe` be `BoolType`, but it looks a lot less weird with the extra copy arg.